niworkflows/data/derivatives.json

```diff
@@ -0,0 +1,149 @@
+      "pattern": "[_/\\\\]+ses-([a-zA-Z0-9]+)",
+      "mandatory": false,
+      "directory": "{subject}{session}"
+    },
+    {
+      "name": "task",
+      "pattern": "[_/\\\\]+task-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "acquisition",
+      "pattern": "[_/\\\\]+acq-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "ceagent",
+      "pattern": "[_/\\\\]+ce-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "reconstruction",
+      "pattern": "[_/\\\\]+rec-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "direction",
+      "pattern": "[_/\\\\]+dir-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "run",
+      "pattern": "[_/\\\\]+run-0*(\\d+)",
+      "dtype": "int"
+    },
+    {
+      "name": "proc",
+      "pattern": "[_/\\\\]+proc-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "modality",
+      "pattern": "[_/\\\\]+mod-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "echo",
+      "pattern": "[_/\\\\]+echo-([0-9]+)"
+    },
+    {
+      "name": "recording",
+      "pattern": "[_/\\\\]+recording-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "space",
+      "pattern": "[_/\\\\]+space-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "suffix",
+      "pattern": "[._]*([a-zA-Z0-9]*?)\\.[^/\\\\]+$"
+    },
+    {
+      "name": "scans",
+      "pattern": "(.*\\_scans.tsv)$"
+    },
+    {
+      "name": "fmap",
+      "pattern": "(phasediff|magnitude[1-2]|phase[1-2]|fieldmap|epi)\\.nii"
+    },
+    {
+      "name": "datatype",
+      "pattern": "[/\\\\]+(func|anat|fmap|dwi|meg|eeg)[/\\\\]+"
+    },
+    {
+      "name": "extension",
+      "pattern": "[._]*[a-zA-Z0-9]*?\\.([^/\\\\]+)$"
+    },
+    {
+      "name": "atlas",
+      "pattern": "[_/\\\\]+atlas-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "roi",
+      "pattern": "[_/\\\\]+roi-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "label",
+      "pattern": "[_/\\\\]+label-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "desc",
+      "pattern": "[_/\\\\]+desc-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "from",
+      "pattern": "(?:^|_)from-([a-zA-Z0-9]+).*xfm"
+    },
+    {
+      "name": "to",
+      "pattern": "(?:^|_)to-([a-zA-Z0-9]+).*xfm"
+    },
+    {
+      "name": "mode",
+      "pattern": "(?:^|_)mode-(image|points)xfm"
+    },
+    {
+      "name": "hemi",
+      "pattern": "hemi-(L|R)"
+    },
+    {
+      "name": "model",
+      "pattern": "model-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "subset",
+      "pattern": "subset-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "resolution",
+      "pattern": "res-([a-zA-Z0-9]+)"
+    },
+    {
+      "name": "density",
+      "pattern": "res-([a-zA-Z0-9]+)"
+    }
+  ],
+  "default_path_patterns": [
+    "sub-{subject}[/ses-{session}]/{datatype<anat>|anat}/sub-{subject}[_ses-{session}][_acq-{acquisition}][_ce-{ceagent}][_rec-{reconstruction}][_space-{space}][_res-{resolution}][_desc-{desc}]_{suffix<T1w|T2w|T1rho|T1map|T2map|T2star|FLAIR|FLASH|PDmap|PD|PDT2|dseg|inplaneT[12]|angio>}.{extension<nii|nii.gz|json>|nii.gz}",
```

Comment:
I don't know if BEP001-compatibility is a priority here, but the current proposed suffix for T2* maps is `T2starmap`, not `T2star`.